### PR TITLE
replacing go.uber.org/atomic by sync/atomic

### PR DESCRIPTION
**Pull request by patil-kshitij**
replacing go.uber.org/atomic by sync/atomic
wrt issue #7160

### CODE REVIEW

**Comment by patil-kshitij** [comment]
Can someone, please re-approve workflows ?
added yaml file in .chloggen directory.

File: .golangci.yml [hunk 1]
```diff
@@ -104,7 +104,6 @@ linters-settings:
     include-go-root: true
     packages-with-error-message:
       # See https://github.com/open-telemetry/opentelemetry-collector/issues/5200 for rationale
-      - sync/atomic: "Use go.uber.org/atomic instead of sync/atomic"
```
Comment (bogdandrutu) [inline .golangci.yml:107]: I think we should disallow go.uber.org/atomic now.

**Review by bogdandrutu — COMMENTED** [review]
I was expecting some go.mod changes as well, since we remove direct usage the uber/atomic may become indirect.

File: .golangci.yml [hunk 2]
```diff
@@ -104,7 +104,6 @@ linters-settings:
     include-go-root: true
     packages-with-error-message:
       # See https://github.com/open-telemetry/opentelemetry-collector/issues/5200 for rationale
```
Comment (Aneurysm9) [inline .golangci.yml:106]:
```suggestion
```

This comment would be misleading if not removed.

File: .golangci.yml (hunk 1 above)
Comment (patil-kshitij) [reply .golangci.yml:107]: Done

File: .golangci.yml (hunk 2 above)
Comment (patil-kshitij) [reply .golangci.yml]: removed that comment

**Comment by patil-kshitij** [comment]
> I was expecting some go.mod changes as well, since we remove direct usage the uber/atomic may become indirect.

@bogdandrutu ,
did "go mod tidy" and changes are now updated in go.mod files.

Thankyou :)

**Comment by patil-kshitij** [comment]
@Aneurysm9 @bogdandrutu ,
Can you please review ?

**Review by Aneurysm9 — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Some race condition in the PRs. Please rebase and fix the usage.

**Comment by patil-kshitij** [comment]
@bogdandrutu , synced this with main branch.
Can you please review ?

**Comment by patil-kshitij** [comment]
Anything pending on this from my side ?

File: processor/memorylimiterprocessor/memorylimiter.go [hunk 3]
```diff
@@ -20,9 +20,9 @@ import (
 	"fmt"
 	"runtime"
 	"sync"
+	"sync/atomic"
```
Comment (codeboten) [inline processor/memorylimiterprocessor/memorylimiter.go:23]: Shouldn't this change the processor's go.mod file as well to remove the direct dependency?

File: exporter/otlpexporter/otlp_test.go [hunk 4]
```diff
@@ -20,12 +20,12 @@ import (
 	"path/filepath"
 	"runtime"
 	"sync"
+	"sync/atomic"
```
Comment (codeboten) [inline exporter/otlpexporter/otlp_test.go:23]: Shouldn't this change the exporter's go.mod file as well to remove the direct dependency?

Comment (patil-kshitij) [reply exporter/otlpexporter/otlp_test.go:23]: Done

File: processor/memorylimiterprocessor/memorylimiter.go (hunk 3 above)
Comment (patil-kshitij) [reply processor/memorylimiterprocessor/memorylimiter.go:23]: Done

**Comment by patil-kshitij** [comment]
Hi @codeboten @bogdandrutu ,
Can you please approve workflows and review PR ?

**Comment by patil-kshitij** [comment]
Hello,
Waiting for some review.
Kindly review, as waiting on this from very long time.
cc - @bogdandrutu @codeboten

**Review by bogdandrutu — APPROVED** [review]

**Comment by patil-kshitij** [comment]
@codeboten , Can you please PR ?

**Review by codeboten — APPROVED** [review]

**Pull request merged**